homeassistant/components/sonos/switch.py

```diff
@@ -0,0 +1,226 @@
+"""Entity representing a Sonos battery level."""
+from __future__ import annotations
+
+import datetime
+import logging
+
+from pysonos.alarms import Alarm
+from pysonos.exceptions import SoCoUPnPException
+
+from homeassistant.components.switch import ENTITY_ID_FORMAT, SwitchEntity
+from homeassistant.const import ATTR_TIME
+from homeassistant.helpers import device_registry as dr, entity_registry as er
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from . import SonosData
+from .const import (
+    DATA_SONOS,
+    DOMAIN as SONOS_DOMAIN,
+    SONOS_ALARM_UPDATE,
+    SONOS_CREATE_ALARM,
+)
+from .entity import SonosEntity
+from .speaker import SonosSpeaker
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_DURATION = "duration"
+ATTR_ID = "alarm_id"
+ATTR_PLAY_MODE = "play_mode"
+ATTR_RECURRENCE = "recurrence"
+ATTR_SCHEDULED_TODAY = "scheduled_today"
+ATTR_VOLUME = "volume"
+ATTR_INCLUDE_LINKED_ZONES = "include_linked_zones"
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up Sonos from a config entry."""
+
+    async def _async_create_entity(speaker: SonosSpeaker) -> None:
+        data_sonos = hass.data[DATA_SONOS]
+
+        for alarm in speaker.available_alarms:
+            # pylint: disable=protected-access
+            if alarm._alarm_id not in data_sonos.alarms:
+                # pylint: disable=protected-access
+                _LOGGER.debug("Creating alarm with id %s", alarm._alarm_id)
+                entity = SonosAlarmEntity(alarm, data_sonos, speaker)
+                async_add_entities([entity])
+                # pylint: disable=protected-access
```

Comment:
We should avoid this and open up the API as public instead.

Reply:
Yes that makes totally sense! How should we proceed?

Reply:
By making that happen? 😄

Reply:
Haha :D I see. I will open up a PR at the API.

Reply:
Did that: https://github.com/home-assistant/core/pull/50719

Reply:
Fixed by #50798 and upcoming commit.